core/zip/src/ZInflate.c

```diff
@@ -1183,7 +1199,7 @@ void R__unzip(int *srcsize, uch *src, int *tgtsize, uch *tgt, int *irep)
   /*   D E C O M P R E S S   D A T A  */
 
   /* New zlib format */
-  if (src[0] == 'Z' && src[1] == 'L') {
+  if (is_valid_header_zlib(src)) {
```

Comment:
That's simply the header, as defined by `core/zip/src/Bits.h`.  The header appears to be custom to ROOT.  For LZMA and LZ4, we have just used a two-byte header.